tests/asgi/_asgi_test_app.py

```diff
@@ -111,6 +112,24 @@ async def emit():
         resp.sse = emit()
 
 
+class Multipart:
+    async def on_post(self, req, resp):
+        parts = {}
+
+        form = await req.get_media()
+        async for part in form:
+            sha1 = hashlib.sha1()
```

Comment:
Another nitpick. Probably we could use a not discouraged hashing algorithm? Like sha256/sha3_256 (not sure if sha3 is in py35)

Reply:
sha3 is py3.6+ only. sha256 is probably ok

Reply:
We are not claiming any cryptographic context here though.
SHA1 has excellent collision properties for other purposes. For instance, Git [_treeish_]( https://git-scm.com/docs/gitglossary#def_SHA1) (commits, and basically any type of reference in Git) is also SHA1, and I don't think that's an issue.

Reply:
Git is old though :). And they are in the process of migrating to a newer / better hash (to sha256 in fact I think).

Reply:
I still don't find think SHA1 is an issue in this example, but maybe I could add a comment stressing that this has nothing to do with cryptographic hashing.

For instance, storage systems use even simpler hashes such CRC, Murmur, MD5 etc for integrity checking.
Hash tables also use performant hashes to spread keys, including Python's `dict` and the `hash()` machinery.

SHA1 is by no means a _discouraged algorithm_. It is discouraged for cryptographic purposes.